js/geo/transform.js

```diff
@@ -346,6 +356,19 @@ Transform.prototype = {
     },
 
     /**
+     * Like `coordinatePoint` but returns the raw vector data instead of just the Point
+     * @param {Coordinate} coord
+     * @returns {Point} screen point
+     * @private
+     */
+    coordinatePoint3d: function(coord) {
+        var scale = this.worldSize / this.zoomScale(coord.zoom);
+        var p = [coord.column * scale, coord.row * scale, 0, 1];
+        vec4.transformMat4(p, p, this.pixelMatrix);
+        return p;
+    },
```

Comment:
This code was copied from `Transform#coordinatePoint`. Can you refactor to eliminate the code duplication?